Limelight/Stream/VideoDecoderRenderer.m

```diff
@@ -378,4 +372,52 @@ - (int)submitDecodeBuffer:(unsigned char *)data length:(int)length bufferType:(i
     return DR_OK;
 }
 
+- (OSStatus) decodeFrameWithSampleBuffer:(CMSampleBufferRef)sampleBuffer frameType:(int)frameType{
+    VTDecodeFrameFlags flags = kVTDecodeFrame_EnableAsynchronousDecompression;
+    
+    return VTDecompressionSessionDecodeFrameWithOutputHandler(decompressionSession, sampleBuffer, flags, NULL, ^(OSStatus status, VTDecodeInfoFlags infoFlags, CVImageBufferRef  _Nullable imageBuffer, CMTime presentationTimestamp, CMTime presentationDuration) {
+        if (status != noErr)
+        {
+            NSError *error = [NSError errorWithDomain:NSOSStatusErrorDomain code:status userInfo:nil];
+            Log(LOG_E, @"Decompression session error: %@", error);
+            LiRequestIdrFrame();
+            return;
+        }
+        
+        if (self->formatDescImageBuffer == NULL || !CMVideoFormatDescriptionMatchesImageBuffer(self->formatDescImageBuffer, imageBuffer)){
+            
+            OSStatus res = CMVideoFormatDescriptionCreateForImageBuffer(kCFAllocatorDefault, imageBuffer, &(self->formatDescImageBuffer));
+            if (res != noErr){
+                Log(LOG_E, @"Failed to create video format description from imageBuffer");
+                return;
+            }
+        }
+        
+        CMSampleBufferRef sampleBuffer;
+        CMSampleTimingInfo sampleTiming = {kCMTimeInvalid, presentationTimestamp, presentationDuration};
+        
+        OSStatus err = CMSampleBufferCreateReadyWithImageBuffer(kCFAllocatorDefault, imageBuffer, self->formatDescImageBuffer, &sampleTiming, &sampleBuffer);
+        
+        if (err != noErr){
+            Log(LOG_E, @"Error creating sample buffer for decompressed image buffer %d", (int)err);
+            return;
+        }
+        
+        // Enqueue the next frame
+        [self->displayLayer enqueueSampleBuffer:sampleBuffer];
```

Comment:
I would add a flush before the [self->displayLayer enqueueSampleBuffer:sampleBuffer] function.
``` 
if (![self->displayLayer isReadyForMoreMediaData]) {
        [self->displayLayer flush];
}
```
Sometimes it happens that not all data is played, and the buffer fills up, so the playback stops.